kernel/arch/dreamcast/hardware/cdrom.c

```diff
@@ -282,32 +315,10 @@ int cdrom_reinit(void) {
 
 /* Enhanced cdrom_reinit, takes the place of the old 'sector_size' function */
 int cdrom_reinit_ex(int sector_part, int cdxa, int sector_size) {
-    int r = -1;
-    int timeout;
-
-    /* Try a few times; it might be busy. If it's still busy
-       after this loop then it's probably really dead. */
-    timeout = 10 * 1000 / 20; /* 10 second timeout */
-
-    while(timeout > 0) {
-        r = cdrom_exec_cmd(CMD_INIT, NULL);
-
-        if(r == 0) break;
-
-        if(r == ERR_NO_DISC || r == ERR_SYS) {
-            return r;
-        }
+    int r;
+    r = cdrom_exec_cmd_timed(CMD_INIT, NULL, 10000);
```

Comment:
Just checking here, since it seems this is a bit of a behavior change... This has been adequately tested on the hardware, right?

If I understand the code correctly, the old path would resubmit the INIT command each pass through its loop if it didn't get a response back, whereas the new version only submits the command one time and then could wait up to 10 seconds...

Reply:
This was written incorrectly.
The drive takes a long time to execute a command when there is no disk, rather than aborting commands with an error. Here it only works because the INIT command never returned an error, since the system calls were just used by the BIOS (in the case without a mod).
Here it is generally forbidden to call INIT without a syscall reset, which was done only by the BIOS, I just added it.
I tested both with and without disk on real hardware and emulators.

Reply:
In any case, there is nothing to worry about if INIT fails here, since it will be performed again if something changes in the status when accessing the file system. Here it can be removed altogether to be honest.

Reply:
As proof that it has never worked before - incorrect use of the abort command syscall :)

Reply:
Ok, Thanks for the confirmation. :+1: